internal/receiver/smartagentreceiver/config.go

```diff
@@ -131,6 +125,19 @@ func (cfg *Config) Unmarshal(componentParser *config.Parser) error {
 	return nil
 }
 
+func recursivelyCapitalizeConfigKeys(settings map[string]interface{}, yamlTags map[string]string) {
```

Comment:
Since this is a viper-centric issue I wonder if `yamlTagsFromStruct` and `recursivelyCapitalizeConfigKeys` should be in a parent function like `respectYamlTagsInAllSettings()`

Reply:
@rmfitzpatrick - I've moved these methods to a helper and exposed `RespectYamlTagsInAllSettings` which can be used by both the receiver and the extension. Let me know if you think it's better.

Reply:
I do indeed!